### PR TITLE
Add Hibernate Pool DAO

### PR DESCRIPTION
**Pull request by apmasell**
JIRA: GLT-1288

### CODE REVIEW

File: sqlstore/src/main/resources/db/migration/V8900__hibernatize_everything.sql [hunk 1]
```diff
@@ -22,7 +22,25 @@ INSERT INTO ProjectOverview_Sample(projectOverview_overviewId, sample_sampleId)
...
+  pool_poolId, elementId FROM Pool_Elements
+  WHERE elementType = 'uk.ac.bbsrc.tgac.miso.core.data.impl.LibraryDilution';
+INSERT INTO Pool_Dilution(pool_poolId, dilution_dilutionId) SELECT
+  pool_poolId, dilution_dilutionId
+  FROM Pool_Elements JOIN emPCRDilution ON Pool_Elements.elementId = emPCRDilution.dilutionId
+  JOIN emPCR ON emPCR.pcrId = emPCRDilution.emPCR_pcrId
+  WHERE elementType = 'uk.ac.bbsrc.tgac.miso.core.data.impl.emPCRDilution'
+  AND NOT EXISTS (SELECT * FROM Pool_Dilution WHERE Pool_Dilution.pool_poolId = Pool_Elements.pool_poolId AND Pool_Dilution.dilution_dilutionId = emPCR.dilution_dilutionId);
+
+DROP TABLE Pool_Elements;
```
Comment (callunity) [inline sqlstore/src/main/resources/db/migration/V8900__hibernatize_everything.sql]: UPDATE Pool SET platformType = UPPER(platformType);

File: sqlstore/src/main/resources/db/migration/V8900__hibernatize_everything.sql [hunk 2]
```diff
@@ -22,7 +22,25 @@ INSERT INTO ProjectOverview_Sample(projectOverview_overviewId, sample_sampleId)
...
+
+INSERT INTO Pool_Dilution(pool_poolId, dilution_dilutionId) SELECT
+  pool_poolId, elementId FROM Pool_Elements
+  WHERE elementType = 'uk.ac.bbsrc.tgac.miso.core.data.impl.LibraryDilution';
+INSERT INTO Pool_Dilution(pool_poolId, dilution_dilutionId) SELECT
+  pool_poolId, dilution_dilutionId
+  FROM Pool_Elements JOIN emPCRDilution ON Pool_Elements.elementId = emPCRDilution.dilutionId
+  JOIN emPCR ON emPCR.pcrId = emPCRDilution.emPCR_pcrId
+  WHERE elementType = 'uk.ac.bbsrc.tgac.miso.core.data.impl.emPCRDilution'
+  AND NOT EXISTS (SELECT * FROM Pool_Dilution WHERE Pool_Dilution.pool_poolId = Pool_Elements.pool_poolId AND Pool_Dilution.dilution_dilutionId = emPCR.dilution_dilutionId);
```
Comment (callunity) [inline sqlstore/src/main/resources/db/migration/V8900__hibernatize_everything.sql]: I thought I had understood you as saying that this code should add the emPCR's parent LibraryDilution to the pool, but I'm seeing the emPCR's ID being added to the pool. Is there something I'm missing?

File: sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/sqlstore/util/AbstractHibernateDao.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package uk.ac.bbsrc.tgac.miso.sqlstore.util;
+
+import org.hibernate.Criteria;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public abstract class AbstractHibernateDao {
```
Comment (callunity) [inline sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/sqlstore/util/AbstractHibernateDao.java]: Are the other Hibernate DAOs supposed to extend this? It's currently a dead class.

File: sqlstore/src/main/resources/db/migration/V8900__hibernatize_everything.sql (hunk 2 above)
Comment (apmasell) [inline sqlstore/src/main/resources/db/migration/V8900__hibernatize_everything.sql]: I've tried to make this clearer, but `emPCR.dilution_dilutionId` is the library dilution from which the emPCR is made while `emPCRDilution.dilutionId` is the ID of the emPCR dilution itself.

**Comment by apmasell** [comment]
These fixes and more.

**Comment by apmasell** [comment]
See https://github.com/TGAC/miso-lims/pull/628

File: sqlstore/src/main/resources/db/migration/V8900__hibernatize_everything.sql (hunk 2 above)
Comment (callunity) [inline sqlstore/src/main/resources/db/migration/V8900__hibernatize_everything.sql]: Aha -- for once, the columns aren't ambiguously-named so they don't need to be aliased. I missed the part where `dilution_dilutionId` came from the emPCR table.

**Pull request closed without merging**